src/ecdsa/ellipticcurve.py

```diff
@@ -92,14 +92,16 @@ def __init__(self, p, a, b, h=None):
             self.__h = h
 
     def __eq__(self, other):
-        if isinstance(other, CurveFp):
-            """Return True if the curves are identical, False otherwise."""
-            return (
-                self.__p == other.__p
-                and self.__a == other.__a
-                and self.__b == other.__b
-            )
-        return NotImplemented
```

Comment:
it's intentionally returning `NotImplemented` in case the other is not CurveFp, as then python will execute `__eq__` on `other`, which in turn can have support to coerce CurveFp into its format and correctly compare them (something we actually had to do with Point and PointJacobi already)

Reply:
Makes sense. Reverted to the original code.